app/src/main/java/org/sopt/and/presentation/ui/auth/SignInActivity.kt

```diff
@@ -0,0 +1,376 @@
+import androidx.compose.foundation.layout.width
+import androidx.compose.foundation.shape.RoundedCornerShape
+import androidx.compose.material3.Button
+import androidx.compose.material3.ButtonDefaults
+import androidx.compose.material3.SnackbarDuration
+import androidx.compose.material3.SnackbarHost
+import androidx.compose.material3.SnackbarHostState
+import androidx.compose.material3.SnackbarResult
+import androidx.compose.material3.Text
+import androidx.compose.runtime.Composable
+import androidx.compose.runtime.getValue
+import androidx.compose.runtime.mutableStateOf
+import androidx.compose.runtime.remember
+import androidx.compose.runtime.rememberCoroutineScope
+import androidx.compose.runtime.setValue
+import androidx.compose.ui.Alignment
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.draw.clip
+import androidx.compose.ui.graphics.Color
+import androidx.compose.ui.layout.ContentScale
+import androidx.compose.ui.platform.LocalConfiguration
+import androidx.compose.ui.platform.LocalContext
+import androidx.compose.ui.res.painterResource
+import androidx.compose.ui.text.style.TextAlign
+import androidx.compose.ui.tooling.preview.Preview
+import androidx.compose.ui.unit.dp
+import androidx.compose.ui.unit.sp
+import kotlinx.coroutines.launch
+import org.sopt.and.R
+import org.sopt.and.presentation.ui.auth.component.AuthTextField
+import org.sopt.and.presentation.ui.main.MainActivity
+import org.sopt.and.presentation.utils.AuthValidation
+import org.sopt.and.presentation.utils.showToast
+import org.sopt.and.ui.theme.ANDANDROIDTheme
+
+class SignInActivity : ComponentActivity() {
+    private lateinit var userEmail: String
+    private lateinit var userPassword: String
+
+    override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+        getUserInfo()
+        enableEdgeToEdge()
+        setContent {
+            ANDANDROIDTheme {
+                SignInScreen(
+                    userEmail = userEmail,
+                    userPassword = userPassword,
+                    navigateToSignUp = { navigateToSignUp() },
+                    navigateToMain = { userEmail, userPassword ->
+                        navigateToMain(
+                            userEmail,
+                            userPassword
+                        )
+                    }
+                )
+            }
+        }
+    }
+
+    private fun getUserInfo() {
+        userEmail = intent.getStringExtra("email") ?: ""
+        userPassword = intent.getStringExtra("password") ?: ""
+    }
+
+    private fun navigateToSignUp() {
+        val intent = Intent(this, SignUpActivity::class.java)
+        startActivity(intent)
+    }
+
+    private fun navigateToMain(userEmail: String, userPassword: String) {
+        val intent = Intent(this, MainActivity::class.java).apply {
+            putExtra("email", userEmail)
+            putExtra("password", userPassword)
+        }
+        startActivity(intent)
+    }
```

Comment:
```suggestion
    private fun navigateToMain(userEmail: String, userPassword: String) {
        startActivity(
            Intent(this, MainActivity::class.java).apply {
                putExtra("email", userEmail)
                putExtra("password", userPassword)
            }
        )
    }
```
저는 이런식으로 변수 생성을 최소화하는 편입니다! 물론 코드 스타일은 취향~

Reply:
intent라는 동일한 이름의 변수가 여러 파일에서 만들어질 필요도 없어지고 깔끔한 것 같습니다! 반영하겠습니다 ^^